backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/ModelValidationEngine.java

```diff
@@ -691,21 +695,29 @@ public void removeModelChange(String tableName, IModelInterceptor interceptor)
 
 	public void fireModelChange(@NonNull final PO po, final ModelChangeType changeType)
 	{
-		final PerformanceMonitoringService performanceMonitoringService = SpringContextHolder.instance.getBeanOr(PerformanceMonitoringService.class, NoopPerformanceMonitoringService.INSTANCE);
-		final String tableName = po.get_TableName();
-		final String changeTypeStr = changeType.toString();
-
-		performanceMonitoringService.monitor(
-				() -> fireModelChange0(po, changeType),
-				Metadata
-						.builder()
-						.name("ModelValidationEngine")
-						.type(Type.MODEL_INTERCEPTOR)
-						.action("fireModelChange")
-						.label("changeType", changeTypeStr)
-						.label("tableName", tableName)
-						.label(PerformanceMonitoringService.LABEL_RECORD_ID, Integer.toString(po.get_ID()))
-						.build());
+		final boolean perfMonIsActive = SYS_CONFIG_BL.getBooleanValue(PERF_MON_SYSCONFIG_NAME, SYS_CONFIG_DEFAULT_VALUE);
+		if(!perfMonIsActive)
+		{
+			fireModelChange0(po, changeType);
+		}
+		else
+		{
+			final PerformanceMonitoringService performanceMonitoringService = SpringContextHolder.instance.getBeanOr(PerformanceMonitoringService.class, NoopPerformanceMonitoringService.INSTANCE);
```

Comment:
check if we can have performanceMonitoringService as a class field? (even if lazy)